### PR TITLE
Add Spiller Aggregate Benchmark

### PR DESCRIPTION
**Pull request by duanmeng**
This PR adds a spiller aggregate benchmark (`AggregateInput` type) based
on the current spiller bencharm framework, and makes a small refactor.

### CODE REVIEW

**Comment by duanmeng** [comment]
@xiaoxmeng Could you please help review? Thanks a lot.

File: velox/exec/tests/SpillerAggregateInputBenchmarkTest.cpp [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  spiller_->setPartitionsSpilled({0});
+  writeSpillData();
+}
+
+void SpillerAggregateInputBenchmarkTest::run() {
+  MicrosecondTimer timer(&executionTimeUs_);
+  spiller_->spill(0, 0);
+}
+
+void SpillerAggregateInputBenchmarkTest::writeSpillData() {
```
Comment (xiaoxmeng) [inline velox/exec/tests/SpillerAggregateInputBenchmarkTest.cpp:92]:
```
s/SpillerAggregateInputBenchmarkTest/SpillerAggregateBenchmarkTest/
```
We can extend to support collecting stats on output? Also consider to override the print stats to log the detailed execution time? As for now, we only collect the time spent on aggregation input spill? Later, we can collect the time spent on output processing. We can skip collecting the time spent on writer spill data? Thanks!

**Review by xiaoxmeng — COMMENTED** [review]
@duanmeng overall looks good. Thanks for working on this!

File: velox/exec/tests/SpillerAggregateInputBenchmarkTest.cpp (hunk 1 above)
Comment (duanmeng) [reply velox/exec/tests/SpillerAggregateInputBenchmarkTest.cpp:92]: Got it. I used to think we needed to add two types of aggregate spill benchmarks, input type and output type. So I added `SpillerAggregateInputBenchmarkTest` to collect the time spent on aggregation input spill, and planned to add the output type of spill benchmark later.
Could I add `SpillerAggregateBenchmarkTest` with only input spill metrics in this PR, and extend it to support both types of spill benchmark and metrics collection in a follow-up PR?

File: velox/exec/tests/SpillerAggregateInputBenchmarkTest.cpp [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+ * limitations under the License.
+ */
+
+#include "velox/exec/tests/SpillerAggregateInputBenchmarkTest.h"
+#include "velox/serializers/PrestoSerializer.h"
+
+#include <gflags/gflags.h>
+#include <deque>
+
+DEFINE_string(
```
[the remark this answers is not in the document]
Comment (duanmeng) [reply velox/exec/tests/SpillerAggregateInputBenchmarkTest.cpp]: Fixed

File: velox/exec/tests/SpillerAggregateInputBenchmarkTest.cpp [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+      rowContainer_->keyTypes().size(),
+      std::vector<CompareFlags>{},
+      fmt::format("{}/{}", spillDir_, FLAGS_spiller_benchmark_name),
+      FLAGS_spiller_benchmark_max_spill_file_size,
+      FLAGS_spiller_benchmark_write_buffer_size,
+      FLAGS_spiller_benchmark_min_spill_run_size,
+      stringToCompressionKind(FLAGS_spiller_benchmark_compression_kind),
+      memory::spillMemoryPool(),
+      executor_.get());
+  spiller_->setPartitionsSpilled({0});
```
[the remark this answers is not in the document]
Comment (duanmeng) [reply velox/exec/tests/SpillerAggregateInputBenchmarkTest.cpp]: Removed

File: velox/exec/tests/SpillerJoinInputBenchmarkTest.cpp [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+using namespace facebook::velox::common;
+using namespace facebook::velox::memory;
+using namespace facebook::velox::exec;
+
+namespace facebook::velox::exec::test {
+namespace {
+const int numSampleVectors = 100;
+} // namespace
+
+void JoinSpillInputBenchmarkTest::setUp() {
```
[the remark this answers is not in the document]
Comment (duanmeng) [reply velox/exec/tests/SpillerJoinInputBenchmarkTest.cpp]: Got it, add JoinSpillInputBenchmarkBase and AggregateSpilltBenchmarkBase, and only implement the main function in SpillerJoinInputBenchmarkTest.cpp and SpillerAggregateBenchmarkTest.cpp.

File: velox/exec/tests/AggregateSpillBenchmarkBase.cpp [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    2 << 30,
+    "The max spill file size");
+DEFINE_uint32(
+    spiller_benchmark_spill_vector_size,
+    100,
+    "The number of rows per each spill vector");
+DEFINE_uint32(
+    spiller_benchmark_spill_dependent_key_num,
+    2,
+    "The number of aggregation dependent key");
```
Comment (xiaoxmeng) [inline velox/exec/tests/AggregateSpillBenchmarkBase.cpp:37]:
```
spiller_benchmark_num_key_columns
The number of key columns
```

**Review by xiaoxmeng — APPROVED** [review]
@duanmeng LGTM % minors. Thanks!

File: velox/exec/tests/AggregateSpillBenchmarkBase.cpp (hunk 5 above)
Comment (duanmeng) [reply velox/exec/tests/AggregateSpillBenchmarkBase.cpp]: Fixed

**Pull request closed without merging**